script/dump_stats.rb

```diff
@@ -0,0 +1,116 @@
+    next unless ENV['QUICK'].nil?
+
+    warnings_multi[work&.warnings&.pluck(:id)&.sort&.join('+')] += 1
+    categories_multi[work&.category_ids&.sort&.join('+')] += 1
+    work&.fandoms&.pluck(:id)&.each do |id|
+      fandoms [id] += 1
+    end
+    work&.ratings&.pluck(:id)&.each do |id|
+      ratings [id] += 1
+    end
+    work&.warnings&.pluck(:id)&.each do |id|
+      warnings [id] += 1
+    end
+    work&.characters&.pluck(:id)&.each do |id|
+      characters [id] += 1
+    end
+    work&.freeforms&.pluck(:id)&.each do |id|
+      freeforms [id] += 1
+    end
+    work&.relationship_ids&.each do |id|
+      relationships [id] += 1
+    end
+    work&.category_ids&.each do |id|
+      categories [id] += 1
+    end
+  end
+  next unless ENV['QUICK'].nil?
+
+  stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},works,all_works,#{w.count}"
+  categories.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},category,#{id},#{value}"
+  end
+  categories_multi.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},category_multi,#{id},#{value}"
+  end
+  warnings.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},warning,#{id},#{value}"
+  end
+  warnings_multi.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},warning_multi,#{id},#{value}"
+  end
+  fandoms.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},fandom,#{id},#{value}"
+  end
+  ratings.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},rating,#{id},#{value}"
+  end
+  characters.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},characters,#{id},#{value}"
+  end
+  freeforms.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},freeform,#{id},#{value}"
+  end
+  relationships.each do |id, value|
+    stats_output.puts "#{days},#{end_date},#{start_date},#{start_date.strftime('%A')},relationships,#{id},#{value}"
+  end
+  stats_output.flush
+
+  days += 1
+  end_date = start_date
+  start_date -= 1.day
+  break if end_date < first_time
+end
+work_output.close
+stats_output.close unless ENV['QUICK'].nil?
+tag_output = File.open(ENV['TAGS_FILENAME'] || '/tmp/tags.csv', 'w')
+tag_output.puts "id,type,name,canonical,cached_count,merger_id\n"
+Tag.find_in_batches do |batch|
+  batch.each do |tag|
+    tag_output.puts "#{tag.id},#{tag.type}," \
+    "#{
+    if tag.taggings_count_cache <= (ENV['TAGS_REDACTED_COUNT'] || 5) && !tag.canonical
+      "Redacted"
+    else
+      tag.name
```

Comment:
Since tag names can include quotes, and quotes are a special character in CSV files, I think you either need to escape the tag name in some way, or [use a library designed for handling CSV](https://ruby-doc.org/stdlib-2.6.1/libdoc/csv/rdoc/CSV.html).